mat_dp_pipeline/sdf/create_sdf.py

```diff
@@ -69,13 +69,15 @@ def create_sdf(
                 # sdf load
                 metadata = SDFMetadata()
                 if main_label is not None:
-                    metadata.main_label = (
-                        main_label
-                        if isinstance(main_label, str)
-                        else main_label.main_label
-                    )
-                elif has_single_target_source:
-                    metadata.main_label = targets.main_label
+                    if isinstance(main_label, str):
+                        metadata.main_label = main_label
+                    elif main_label is not None and main_label.main_label is not None:
```

Comment:
`main_label is not None` is True for sure given line 71

Reply:
yep